### PR TITLE
mg_printf formatting

### PR DESCRIPTION
**Pull request by hansipie**
Wrong format parameter for an integer.

### CODE REVIEW

**Comment by bel2125** [comment]
There are some merge conflicts that do not allow automatic merging.
I will have a look, but I think it's easier to change the four code blocks manually than to merge the nine commits.

File: test/unit_test.c [hunk 1]
```diff
@@ -1147,7 +1147,7 @@ static int request_test_handler(struct mg_connection *conn, void *cbdata)
 	          "Content-Type: text/plain\r\n\r\n");
 
 	for (i = 0; i < 20; i++) {
-		mg_printf(conn, "%s\r\n", i);
+		mg_printf(conn, "%d\r\n", i);
```
Comment (bel2125) [inline test/unit_test.c]: This must be %x here anyway.

This part of the code is already converted to the "check" unit test framework: 
https://github.com/civetweb/civetweb/blob/master/test/public_server.c#L323

File: src/civetweb.c [hunk 2]
```diff
@@ -6079,6 +6090,10 @@ static char *addenv(struct cgi_env_block *block, const char *fmt, ...)
 	char *added;
 	va_list ap;
 
+	if (env == NULL) {
+		return;
+	}
+
 	if (block == NULL || fmt == NULL) {
```
Comment (bel2125) [inline src/civetweb.c]: This line (`if (block == NULL || fmt == NULL) {`) does not exist anymore in the current source.

File: src/civetweb.c [hunk 3]
```diff
@@ -6079,6 +6090,10 @@ static char *addenv(struct cgi_env_block *block, const char *fmt, ...)
 	char *added;
 	va_list ap;
 
+	if (env == NULL) {
```
Comment (bel2125) [inline src/civetweb.c]: The local variables here are `added` and `ap`, the parameters `block` and `fmt`.
There is no variable `env` in the code shown here.

There is an `env` parameter in the current code - but the `addenv` function has been completely rewritten between the version with the variables shown here (added, ap, fmt and block) and the current version (with variables n, space, truncated, added, ap, env and fmt).

I think this patch can no longer be applied to the current version.

**Comment by bel2125** [comment]
It was not possible to merge this pull request.
The patch to `unit_test.c` has been applied, but I completely rewrote `addenv` in the meantime, so this patch can no longer be applied. I think it's no longer required in the current version, but please have a look.

**Comment by hansipie** [comment]
Is ok for me. You can refuse the pull request and change the four code blocks manually, no PB.
I'll take care of the conflict on my side.

**Pull request closed without merging**